hostfactory/host_provider/src/cyclecloud_provider.py

```diff
@@ -379,111 +379,114 @@ def _create_status(self, input_json, output_handler=None, update_completed_nodes
+                                    logger.warning("_create_status: Node does not have hostname using %s ", hostname)
+                                except Exception:                                
+                                    # We report status as running even though the node is ready, as we don't have a hostname.
+                                    logger.warning("_create_status: No hostname set and could not convert ip %s to hostname for \"%s\" VM.", node.private_ip, node)
+                                    machine_result = MachineResults.executing
+                                    machine_status = MachineStates.building
+                                    request_status = RequestStates.running
+                                    hostname = ""
+                            if hostname:
+                                completed_nodes.append({"hostname": hostname, "nodeid": node_id})
+                    else:
+                        machine_result = MachineResults.executing
+                        machine_status = MachineStates.building
+                        request_status = RequestStates.running
                 
+                except Exception as e:
+                    logger.exception("Error processing node %s with exception %s but reporting machine status as building", node, e)
+                    machine_result = MachineResults.executing
+                    machine_status = MachineStates.building
+                    request_status = RequestStates.running 
                 machine = {
-                    "name": hostname or "",
-                    "status": machine_status,
-                    "result": machine_result,
-                    "machineId": self.cluster.get_node_id(node) or "",
-                    # launchTime is manditory in Symphony
-                    # maybe we can add something so we don"t have to expose this
-                    # node["PhaseMap"]["Cloud.AwaitBootup"]["StartTime"]["$date"]
-                    "launchtime": int(time.time()),
-                   # "launchtime": node.get("LaunchTime") or int(time.time()),
-                    "privateIpAddress": private_ip_address or "",
-                    "message": ""
-                    #"message": node.get("StatusMessage") or ""
-                }
-                
+                        "name": hostname or "",
+                        "status": machine_status,
+                        "result": machine_result,
+                        "machineId": self.cluster.get_node_id(node) or "",
+                        # launchTime is manditory in Symphony
+                        # maybe we can add something so we don"t have to expose this
+                        # node["PhaseMap"]["Cloud.AwaitBootup"]["StartTime"]["$date"]
+                        "launchtime": int(time.time()),
+                       # "launchtime": node.get("LaunchTime") or int(time.time()),
```

Comment:
Let's remove this comment too since it misleadingly uses node.get (and it's really removed code)